src/typescript/frontend/src/components/pages/metadata/MetadataPage.tsx

```diff
@@ -0,0 +1,276 @@
+"use client";
+
+import {
+  MarketMetadataByMarketAddress,
+  MarketProperties,
+  SetMarketProperties,
+} from "@/contract-apis";
+import { AccountAddress } from "@aptos-labs/ts-sdk";
+import { useQuery } from "@tanstack/react-query";
+import { useAptos } from "context/wallet-context/AptosContextProvider";
+import {
+  type ChangeEventHandler,
+  type MouseEventHandler,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+
+const PLACEHOLDERS = new Map([
+  ["Discord", "https://discord.gg/..."],
+  ["Motto", "The best shoe market ever!"],
+  ["Telegram", "https://t.me/..."],
+  ["Website", "https://..."],
+  ["X profile", "https://x.com/..."],
+]);
+
+const DEFAULT_FIELDS = new Map(Array.from(PLACEHOLDERS.entries().map(([key, _]) => [key, ""])));
+
+const BUTTON_CLASSNAME = "text-white px-4 border-ec-blue border-solid border-[3px] rounded-xl";
+const INPUT_CLASSNAME =
+  "text-white px-4 border-ec-blue border-solid border-[3px] rounded-xl bg-black outline-none";
+
+const FormEntry = ({
+  value,
+  onChange,
+  placeholder,
+  name,
+  onClick,
+  buttonText,
+  className,
+  disabled = false,
+}: {
+  value: string;
+  onChange: ChangeEventHandler<HTMLInputElement>;
+  onClick?: MouseEventHandler<HTMLButtonElement>;
+  placeholder: string;
+  name: string;
+  buttonText?: string;
+  className?: string;
+  disabled?: boolean;
+}) => (
+  <div className={`mx-auto relative`}>
+    <div className="max-w-[300px] text-white">{name}</div>
+    <input
+      tabIndex={1}
+      className={`${INPUT_CLASSNAME} w-[300px] ${className}`}
+      value={value}
+      placeholder={placeholder}
+      onChange={onChange}
+    ></input>
+    {buttonText && (
+      <button
+        className={`${BUTTON_CLASSNAME} absolute right-[-60px] ${disabled ? "!border-dark-gray" : ""}`}
+        onClick={onClick}
+        tabIndex={2}
+        disabled={disabled}
+      >
+        {buttonText}
+      </button>
+    )}
+  </div>
+);
+
+const MetadataPage = () => {
+  const [fields, setFields] = useState<Map<string, string>>(DEFAULT_FIELDS);
+  const [marketAddress, setMarketAddress] = useState<string>("");
+  const [addFieldName, setAddFieldName] = useState<string>("");
+  const [pasted, setPasted] = useState(false);
+
+  const { account, aptos, submit } = useAptos();
+
+  const isMarketAddressValid = useMemo(
+    () => AccountAddress.isValid({ input: marketAddress }).valid,
+    [marketAddress]
+  );
+
+  const { data: marketExists } = useQuery({
+    queryKey: ["metadata-market-exists", marketAddress],
+    queryFn: async () => {
+      if (!isMarketAddressValid) {
+        return null;
+      }
+      try {
+        const market = await MarketMetadataByMarketAddress.view({
+          aptos,
+          marketAddress,
+        });
+        return market.vec.length !== 0;
+      } catch (e) {
+        console.error("Could not get if market exists", e);
```

Comment:
```suggestion
        console.error("Failed to call the market metadata view function.", e);
```

Reply:
Fixed in f8a7353b87d46bad1fbfa010e10d2f0ab9ad9176.

I used the function Move path to not confuse the metadata function from the `emojicoin_dot_fun` module with the one from `market_metadata`.